engine.js

```diff
@@ -331,7 +396,65 @@ var hmdajson = require('./lib/hmdajson'),
      * -----------------------------------------------------
      */
 
-    // TODO
+    var resolveArg = function(arg, contextList) {
+        var tokens = arg.split('.');
+        for (var i = 0; i < contextList.length; i++) {
+            var mappedArg = contextList[i];
+
+            for (var j = 0; j < tokens.length; j++) {
+                mappedArg = mappedArg[tokens[j]];
+                if (mappedArg === undefined) {
+                    break;
+                }
+            }
+            
+            if (mappedArg !== undefined) {
+                return mappedArg;
+            }
+        }
+        return arg;
```

Comment:
I think this should throw an exception if the arg can't be resolved, rather than returning the arg. It could result in some funky behavior if a rule isn't written correctly.